ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -0,0 +1,32 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { currentConfirmationSelector } from '../../../../../../selectors';
+import { selectConfirmationAdvancedDetailsOpen } from '../../../../selectors/preferences';
+import { AdvancedDetails } from '../shared/advanced-details/advanced-details';
+import { ApproveDetails } from '../shared/approve-details/approve-details';
+import { GasFeesSection } from '../shared/gas-fees-section/gas-fees-section';
+
+const ApproveInfo = () => {
+  const transactionMeta = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const showAdvancedDetails = useSelector(
+    selectConfirmationAdvancedDetailsOpen,
+  );
+
+  if (!transactionMeta?.txParams) {
+    return null;
+  }
+
+  return (
+    <>
+      <ApproveDetails />
+      <GasFeesSection />
```

Comment:
Rather than duplicating the usage of these sections and components in every transaction type, could we instead just add children to a `BaseTransactionInfo` so we don't have to duplicate the layout and can maintain a single base template that we add rows to?

Reply:
It's not necessarily the amount of lines, but the conceptual duplication in future if we have 5 different types for example, each re-defining this same list. And then we want to add something to all of them.

But agree it's not urgent now.